<?xml version="1.0" ?>
<robot name="m2wr" xmlns:xacro="https://www.ros.org/wiki/xacro" >   <!-- this second parameter is importent to make this file a valid xacro file -->

<!-- in rviz red - x  | green - y | blue -z  -->

<!-- while using odom tf do not forget to do   rosparam set use_sim_time true  as this gives all simulations the same time -->


<xacro:include filename="$(find bug1and2_algorithm)/urdf/material.xacro" />
<xacro:include filename="$(find bug1and2_algorithm)/urdf/gazebo.xacro" />
<xacro:include filename="$(find bug1and2_algorithm)/urdf/link_joint.xacro" />

    
  <link name="link_chassis">
    <!-- pose and inertial -->
    <pose>0 0 0.1 0 0 0</pose>
    
    <inertial>
      <mass value="15"/>
      <origin rpy="0 0 0" xyz="0 0 0.1"/>
      <inertia ixx="0.0395416666667" ixy="0" ixz="0" iyy="0.106208333333" iyz="0" izz="0.106208333333"/>
    </inertial>
    
    <collision name="collision_chassis">
      <geometry>
        <cylinder radius = "0.3" length= "0.07"/>
      </geometry>
    </collision>
    
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder radius = "0.3" length= "0.07"/>
      </geometry>
      <material name="blue"/>
    </visual>
    
    <!-- caster front -->
    <collision name = "caster_front_collision">
      <origin rpy=" 0 0 0" xyz="0.35 0 -0.05"/>
      <geometry>
        <sphere radius="0.05"/>
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu>0</mu>
            <mu2>0</mu2>
            <slip1>1.0</slip1>
            <slip2>1.0</slip2>
          </ode>
        </friction>
      </surface>
    </collision>
    <visual name="caster_front_visual">
      <origin rpy=" 0 0 0" xyz="0.2 0 -0.05"/>
      <geometry>
        <sphere radius="0.05"/>
      </geometry>
    </visual>
    </link>


  <!-- Create sensor link  -->
  <xacro:sensor_link /> 
  <!-- Create sensor joint --> 
  <xacro:sensor_joint /> 

  <!-- Create wheel right -->  
  <xacro:link_wheel name="link_right_wheel" />
  <!-- Create joint right --> 
  <xacro:joint_wheel name="joint_right_wheel" param_xyz="-0.05 0.3 0" child="link_right_wheel"  />
  <!-- Create wheel left --> 
  <xacro:link_wheel name="link_left_wheel" />
  <!-- Create joint left --> 
  <xacro:joint_wheel name="joint_left_wheel" param_xyz="-0.05 -0.3 0" child="link_left_wheel"  />
  
</robot>